yt/frontends/gadget/data_structures.py

```diff
@@ -116,6 +124,11 @@ def __init__(self, filename, dataset_type="gadget_binary",
             return
         self._header_spec = self._setup_binary_spec(
             header_spec, gadget_header_specs)
+        self._header_size = _compute_header_size(self._header_spec)
+        if self._header_size != 256:
+            only_on_root(
+                mylog.warn, "Non-standard header size! (%s instead of 256)",
```

Comment:
Can you add a sentence or two with a little bit of extra information about why this might have happened, e.g. they're using a non-standard version of the gadget output format, they should verify that their data are loaded correctly, and they might also want to customize their field configuration, and maybe even add a link to a relevant place in the docs.

For long multiline strings like that I usually format it like this:

```
only_on_root(mylog.warn,
             "this is a very long message that I'm spreading out over "
             "multiple lines. Note that I ended the previous line with "
             "a space. Also note how there aren't any characters "
             "between the open and end quotes besides newlines.")
```

Reply:
Thanks for the suggestion. I'll make the change.

Reply:
Before I commit the change, does this message look good?
```python
only_on_root(
    mylog.warn,
    "Non-standard header size is detected! "
    "Gadget-2 standard header is 256 bytes, but yours is %s. "
    "Make sure a non-standard header is actually expected. "
    "Otherwise something is wrong, "
    "and you might want to check how the dataset is loaded. "
    "Futher information about header specification can be found in "
    "https://yt-project.org/docs/dev/examining/loading_data.html#header-specification.",
    self._header_size
)
```

Reply:
It looks like you didn't actually use the `%` operator in the string to get the formatting to process, but other than that LGTM.

Reply:
Actually it is used. The format is copied from other occurrences of `only_on_root(mylog.xxx, ...)`. But know I wonder how. It must be some magic in how `mylog` works.